data-collection/src/main/java/refactoringml/RefactoringAnalyzer.java

```diff
@@ -33,49 +29,52 @@
 	private Repository repository;
 	private boolean storeFullSourceCode;
 	private String fileStorageDir;
+	private PMDatabase pmDatabase;
 
 	private static final Logger log = LogManager.getLogger(RefactoringAnalyzer.class);
 
-	public RefactoringAnalyzer (Project project, Database db, Repository repository, String fileStorageDir, boolean storeFullSourceCode) {
+	public RefactoringAnalyzer (Project project, Database db, Repository repository, PMDatabase pmDatabase, String fileStorageDir, boolean storeFullSourceCode) {
 		this.project = project;
 		this.db = db;
 		this.repository = repository;
 		this.storeFullSourceCode = storeFullSourceCode;
-
 		this.tempDir = null;
+		this.pmDatabase = pmDatabase;
 		this.fileStorageDir = lastSlashDir(fileStorageDir);
 	}
 
-	public List<RefactoringCommit> collectCommitData(RevCommit commit, CommitMetaData superCommitMetaData, List<Refactoring> refactoringsToProcess) throws IOException {
+	public List<RefactoringCommit> collectCommitData(RevCommit commit, CommitMetaData superCommitMetaData, List<Refactoring> refactoringsToProcess, List<DiffEntry> entries) {
 		List<RefactoringCommit> allRefactorings = new ArrayList<>();
 
 		try {
-			// get the map between new path -> old path
-			HashMap<String, String> filesMap = getMapWithOldAndNewFiles(repository, commit);
+			//TODO: is this obsolete? JGit misses many files, e.g. after move source dir refactorings
+			//get the map between new path -> old path
+			HashMap<String, String> filesMap = getMapWithOldAndNewFiles(entries);
 
 			// get the map between class names
 			HashMap<String, String> classAliases = getClassAliases(refactoringsToProcess);
-
 			//Iterate over all Refactorings found for this commit
 			for (Refactoring refactoring : refactoringsToProcess) {
+
 				String refactoringSummary = refactoring.toString().trim();
 				log.debug("Process Commit [" + commit.getId().getName() + "] with Refactoring: [" + refactoringSummary + "]");
 
 				//loop over all refactored classes, multiple classes can be refactored by the same refactoring, e.g. Extract Interface Refactoring
 				for (ImmutablePair<String, String> pair : refactoredFilesAndClasses(refactoring, refactoring.getInvolvedClassesBeforeRefactoring())) {
 					// get the name of the file before the refactoring
 					// if the one returned by RMiner exists in the map, we use the one in the map instead
-					String refactoredClassFile = pair.getLeft();
-					if(filesMap.containsKey(refactoredClassFile))
-						refactoredClassFile = filesMap.get(refactoredClassFile);
+					String refactoredClassFile = enforceUnixPaths(pair.getLeft());
+					//ignore the filename from jgit for move source dirs etc, because it is unreliable (#133)
```

Comment:
So, what you are saying here is that, for the Rename refactorings things we should always trust RefactoringMiner, but for Non Rename refactorings, we trust Git.

The problem of not using the JGit one is that, if the file doesn't exist in commit - 1, we will have an exception.

Also remember that there are refactorings in RefactoringMiner that it returns the "new name" twice, instead of the "old name". That's why we always rely on the Git one. 

In which cases did you see JGit being less reliable than RMiner?

Reply:
The problem with JGit:
It misses quite a few rename and move source dir refactorings, because it considers many files as newly introduced, even though they are just renamed files (https://github.com/tsantalis/RefactoringMiner/issues/89, #133).
I only observed these issues for these two types of refactorings and RMiner works good in these cases, thus I trust RMiner instead of JGit.